src/Compilers/Core/Portable/Collections/OrderPreservingMultiDictionary.cs

```diff
@@ -19,10 +22,12 @@ namespace Microsoft.CodeAnalysis.Collections
     /// </remarks>
     internal sealed class OrderPreservingMultiDictionary<K, V> :
         IEnumerable<KeyValuePair<K, OrderPreservingMultiDictionary<K, V>.ValueSet>>
+        where K: notnull
+        where V: notnull
     {
         #region Pooling
 
-        private readonly ObjectPool<OrderPreservingMultiDictionary<K, V>> _pool;
+        private readonly ObjectPool<OrderPreservingMultiDictionary<K, V>>? _pool;
```

Comment:
>? _pool [](start = 73, length = 7)

Is `_pool` ever `null`?

Reply:
Hmm ... maybe not anymore. The `Free` method indicates it believes it could be. Let me dig into that a bit.

Reply:
Confirmed yes it is `null` in the case of the paramterless constructor. 

---
In reply to: [354041108](https://github.com/dotnet/roslyn/pull/40029#discussion_r354041108) [](ancestors = 354041108)